packages/core/parcel-bundler/src/transforms/babel/jsx.js

```diff
@@ -12,6 +12,26 @@ const JSX_PRAGMA = {
   hyperapp: 'h'
 };
 
+/**
+ * Solves a use case where people use JSX in .js files
+ */
+function isUsingJSXinJS(asset) {
+  // matches import * as React from 'react' and alike
+  const es6Candidate = /from\s+[`"'](react|preact|nervjs|hyperapp)[`"']/;
```

Comment:
Would this make sense to be just one regex?

That would simplify this code a lot